localstack/services/stepfunctions/asl/component/state/state_execution/state_task/service/resource.py

```diff
@@ -105,4 +110,16 @@ def __init__(
         )
         self.service_name = service_name
         self.api_name = api_name
-        self.api_action = resource_arn.split(":")[-1]
+
+        arn_parts = resource_arn.split(":")
+        tail_part = arn_parts[-1]
+        tail_parts = tail_part.split(".")
+        self.api_action = tail_parts[0]
```

Comment:
nit: would generally be a good idea in places like this to add comments giving examples (in this case for the `resource_arn`)